mixedbread_ai_haystack/rerankers/reranker.py

```diff
@@ -0,0 +1,127 @@
+
+    docs = [Document(content="Paris"), Document(content="Berlin")]
+    query = "What is the capital of Germany?"
+    output = ranker.run(query=query, documents=docs)
+    docs = output["documents"]
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str = "mixedbread-ai/mxbai-rerank-large-v1",
+        top_k: int = 10,
+        meta_fields_to_rank: Optional[List[str]] = None,
+        **kwargs,
+    ):
+        """
+        Initializes an instance of the 'MixedbreadAIReranker'.
+
+        Parameters:
+            model (str): Mixedbread AI model name.
+            top_k (int): The maximum number of documents to return.
+            meta_fields_to_rank (Optional[List[str]]): List of meta fields that should be concatenated
+                with the document content for reranking.
+        """
+        super(MixedbreadAIReranker, self).__init__(**kwargs)
+
+        self.model = model
+        self.top_k = top_k
+        self.meta_fields_to_rank = meta_fields_to_rank or []
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        Returns:
+            Dict[str, Any]: Dictionary with serialized data.
+        """
+        parent_params = super(MixedbreadAIReranker, self).to_dict()["init_parameters"]
+
+        return default_to_dict(
+            self,
+            **parent_params,
+            model=self.model,
+            top_k=self.top_k,
+            meta_fields_to_rank=self.meta_fields_to_rank,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "MixedbreadAIReranker":
+        """
+        Deserializes the component from a dictionary.
+
+        Parameters:
+            data (Dict[str, Any]): The dictionary to deserialize from.
+
+        Returns:
+            MixedbreadAIReranker: The deserialized component.
+        """
+        return from_dict(cls, data)
+
+    @component.output_types(documents=List[Document], meta=Dict[str, Any])
+    def run(self, query: str, documents: List[Document], top_k: Optional[int] = None) -> Dict[str, Any]:
+        """
+        Uses the Mixedbread Reranker to re-rank the list of documents based on the query.
+
+        Parameters:
+            query (str): Query string.
+            documents (List[Document]): List of Documents to be ranked.
+            top_k (Optional[int]): The maximum number of Documents you want the Ranker to return.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the ranked documents with the following key:
+                - `documents`: List of Documents most similar to the given query in descending order of similarity.
+
+        Raises:
+            ValueError: If `top_k` is not > 0.
+        """
+        top_k = top_k or self.top_k
+        if top_k <= 0:
+            raise ValueError(f"top_k must be > 0, but got {top_k}")
+
+        dicts = [doc.to_dict() for doc in documents]
+        rank_fields = [*self.meta_fields_to_rank, "content"]
```

Comment:
Not sure how this would behave, if there's a "content" in the rank_fields already. The API might validate that they're unique, maybe not. Anyways a set would resolve that